OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -477,8 +481,8 @@ bool CanTargetLocation(Actor self, CPos location, TargetModifiers modifiers, ref
 
 				IsQueued = modifiers.HasModifier(TargetModifiers.ForceQueue);
 
-				// Targeting the terrain is only possible with force-attack modifier
-				if (modifiers.HasModifier(TargetModifiers.ForceMove) || !modifiers.HasModifier(TargetModifiers.ForceAttack))
+				// Targeting the terrain is only possible with force-attack modifier or when TargetsTerrain is set
+				if (modifiers.HasModifier(TargetModifiers.ForceMove) || !(modifiers.HasModifier(TargetModifiers.ForceAttack) || ab.Info.TargetsTerrain))
```

Comment:
perhaps move it as the first check and split the 3 check into newlines

Reply:
I find the one-line version to be more readable, but ok.